docs/pages/machine-id/access-guides/kubernetes.mdx

```diff
@@ -21,238 +21,152 @@ controls that Teleport provides for human users.
+Teleport Proxy attaches the groups configured (using `kubernetes_groups`) in the
+bot's Teleport roles to the request. These groups are then used to configure a
+RoleBinding or ClusterRoleBinding in Kubernetes to grant specific permissions
+within the Kubernetes cluster to the bot.
 
-```yaml
-kind: role
-metadata:
-  name: machine-id-kube-role
-version: v7
-spec:
-  allow:
-    kubernetes_labels:
-      '*': '*'
-    kubernetes_resources:
-      - kind: pod
-        namespace: "*"
-        name: "*"
-    kubernetes_groups:
-    - example
-    kubernetes_users:
-    - alice
-```
-
-Be sure to configure the `allow` fields to match your environment:
- - `kubernetes_labels` must be specified to allow access to a matching
-   Kubernetes cluster in your Teleport environment.
+For the purpose of this guide, we will bind the `editor` group to the default
+`edit` ClusterRole that is preconfigured in most Kubernetes clusters to give
+the bot read and write access to resources in all the cluster namespaces.
 
- - Each item in `kubernetes_resources` must match the namespace and name of a
-   Kubernetes resource that you would like users with `machine-id-kube-role` to
-   access. Currently, Teleport only supports the `pod` kind. In the
-   configuration above, this role can access all pods in all namespaces.
+When configuring this for a production environment, you should consider:
 
- - One (or both) of `kubernetes_groups` and `kubernetes_users` must be specified
-   to so that the bot user may be mapped to a Kubernetes user and/or group.
+- If RoleBinding should be used instead of ClusterRoleBinding to limit the
+  bot's access to a specific namespace.
+- If a Role should be created that grants the bot the least privileges
+  necessary rather than using a pre-existing general Role such as `edit`.
 
-   If only `kubernetes_groups` is set, `kubernetes_users` will be set to the
-   bot's username by default.
+To bind the `editor` group to the `edit` Cluster Role, execute:
 
-   The user and group values entered here are relayed directly to the Kubernetes
-   cluster as incoming user and group names. Kubernetes users and groups do not
-   need to be created in advance, but do need to match one or more
-   `RoleBindings` for the bot user to have any permissions on the target
-   Kubernetes cluster.
-
-Once finished, create the role in Teleport:
 ```code
-$ tctl create -f role.yaml
+$ kubectl create clusterrolebinding teleport-editor-edit \
+  --clusterrole=edit \
+  --group=editor
 ```
 
-<Details
-title="Create a Kubernetes Role and RoleBinding"
-opened={true}
->
-If you don't have an existing Kubernetes `Role` or `RoleBinding` to grant your
-Machine ID bot, create one as follows.
+With the appropriate RoleBinding configured in Kubernetes to grant access to
+a specific group, you now need to add this group to the role that the bot
+will impersonate when producing credentials. You also need to grant the bot
+access through Teleport to the cluster itself.
```

Comment:
Since the Kubernetes-related RBAC options are so complex, I wouldn't include detailed instructions here for configuring a Teleport role that can access a Kubernetes cluster. Instead, I'd make defining a Teleport role for the bot a prerequisite and link to the [how-to guide](https://goteleport.com/docs/kubernetes-access/manage-access/rbac/) and [reference](https://goteleport.com/docs/kubernetes-access/controls/) for Kubernetes RBAC in Teleport. 

We can then make the example role explicitly no-op, telling the reader to replace it with something with actual permissions.

How does that sound?

Reply:
I think until we make it easier to add new roles to an existing bot, it makes it harder to link out to these other guides as those guides will describe creating a new role rather than modifying an existing one. Once that functionality is added, I can come back and further simplify this guide to link off to the k8s access docs as a pre-requisite.